lisa/sut_orchestrator/azure/common.py

```diff
@@ -1232,36 +1232,49 @@ def generate_sas_token(
 
 
 def get_blob_service_client(
-    credential: Any,
-    subscription_id: str,
-    cloud: Cloud,
-    account_name: str,
-    resource_group_name: str,
+    credential: Optional[Any] = None,
+    subscription_id: Optional[str] = None,
+    cloud: Optional[Cloud] = None,
+    account_name: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+    connection_string: Optional[str] = None,
```

Comment:
Is connection string the common concept? should it be a sas url?

Reply:
if it is a sas url, use requests / wget can reach the same goal.

Reply:
I mean if the **connection string** is a common concept in blob service client? If so, it's ok to use it as a variable name.

Reply:
in BlobServiceClient.from_connection_string the parameter called
```
        :param str conn_str:
            A connection string to an Azure Storage account.
```

it can be below two types

```
Complete Connection String
DefaultEndpointsProtocol=https;AccountName=<your_account_name>;AccountKey=<your_account_key>;EndpointSuffix=core.windows.net
```

```
SAS Token Connection String
BlobEndpoint=https://<your_account_name>.blob.core.windows.net/;SharedAccessSignature=<your_sas_token>
```